lib/appmap/handler/rails/template.rb

```diff
@@ -14,16 +14,30 @@ class Template
         # The class name is generated from the template path. The package name is
         # 'app/views', and the method name is 'render'. The source location of the method
         # is, of course, the path to the view template.
-        TemplateMethod = Struct.new(:path) do
-          private_instance_methods :path
+        class TemplateMethod
           attr_reader :class_name
- 
+
+          attr_reader :path
+          private_instance_methods :path
+
           def initialize(path)
-            super
+            @path = path
 
             @class_name = path.parameterize.underscore
           end
-  
+
+          def id
+            [ package, path, name ]
+          end
+
+          def hash
+            id.hash
+          end
+
+          def eql?(other)
+            id.eql? other.id
+          end
```

Comment:
These Method objects are placed into a Set in order to make them unique. To behave properly in a set, `hash` and `eql?` are required.